core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -507,4 +511,18 @@ protected static Fuzziness randomFuzziness(String fieldName) {
     protected static boolean isNumericFieldName(String fieldName) {
         return INT_FIELD_NAME.equals(fieldName) || DOUBLE_FIELD_NAME.equals(fieldName);
     }
+
+    private static class MockTermsLookupFetchService extends TermsLookupFetchService {
+
+        public MockTermsLookupFetchService() {
+            super(null, Settings.Builder.EMPTY_SETTINGS);
+        }
+
+        @Override
+        public List<Object> fetch(TermsLookup termsLookup) {
+            // right now we just spit out the index name, type etc .. as terms
+            return Lists.newArrayList((Object) termsLookup.getIndex(), termsLookup.getType(), termsLookup.getId(),
+                    termsLookup.getRouting(), termsLookup.getPath());
```

Comment:
we are using a mocked service, we can retrieve the generated values through the mocked service I guess. Even pre-generate random values when the service is created. I think that would be already better than returning index, type etc.